### PR TITLE
mpp fully support collator

### PR DESCRIPTION
**Pull request by windtalker**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
MPP has fully support for collator, so delete the mpp restriction when new collation is on.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/use-tiflash.md [hunk 1]
```diff
@@ -389,6 +389,3 @@ TiFlash 目前尚不支持的一些功能，与原生 TiDB 可能存在不兼容
         ```
 
         以上示例中，在 TiDB 和 TiFlash 中，`a/b` 在编译期推导出来的类型都为 `Decimal(7,4)`，而在 `Decimal(7,4)` 的约束下，`a/b` 返回的结果应该为 `0.0000`。但是在 TiDB 中，`a/b` 运行期的精度比 `Decimal(7,4)` 高，所以原表中的数据没有被 `where a/b` 过滤掉。而在 TiFlash 中 `a/b` 在运行期也是采用 `Decimal(7,4)` 作为结果类型，所以原表中的数据被 `where a/b` 过滤掉了。
-
-* TiFlash MPP 模式不支持如下功能：
-    * 在配置项 [`new_collations_enabled_on_first_bootstrap`](/tidb-configuration-file.md#new_collations_enabled_on_first_bootstrap) 的值为 `true` 时，MPP 不支持 join 的连接键类型为字符串或 `group by` 聚合运算时列类型为字符串的情况。在处理这两类查询时，默认不选择 MPP 模式。
```
Comment (TomShawn) [inline tiflash/use-tiflash.md:394]: 这里是否可以写，自 v5.2.0 起 MPP 支持这些 collator 呢？

Comment (windtalker) [reply tiflash/use-tiflash.md:394]: 你是说保留这个限制，然后另外说明从v5.2.0之后没有这个限制了？

Comment (TomShawn) [reply tiflash/use-tiflash.md:394]: 如果这是一个 v5.2 的新功能或功能增强的话，能否单独用文字说明一下，这是个功能增强呢？因为可能是要链到 5.2 release notes 中的。

**Review by TomShawn — APPROVED** [review]
LGTM

**Review by zanmato1984 — APPROVED** [review]
LGTM

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**